pkg/controller/mutate_aks.go

```diff
@@ -13,64 +13,85 @@ import (
 
 	"github.com/Azure/application-gateway-kubernetes-ingress/pkg/annotations"
 	"github.com/Azure/application-gateway-kubernetes-ingress/pkg/appgw"
+	"github.com/Azure/application-gateway-kubernetes-ingress/pkg/azure"
 	"github.com/Azure/application-gateway-kubernetes-ingress/pkg/events"
 	"github.com/Azure/application-gateway-kubernetes-ingress/pkg/k8scontext"
 )
 
+type ipResource string
+type ipAddress string
+
 // MutateAKS applies changes to Kubernetes resources.
-func (c AppGwIngressController) MutateAKS(event events.Event) error {
+func (c AppGwIngressController) MutateAKS() error {
+	appGw, cbCtx, err := c.getAppGw()
+	if err != nil {
+		return err
+	}
+
+	// update all ingresses with IP address obtained from existing App Gateway configuration
+	for _, ingress := range cbCtx.IngressList {
+		c.updateIngressStatus(appGw, cbCtx, ingress)
+	}
 	return nil
 }
 
-func (c AppGwIngressController) updateIngressStatus(appGw *n.ApplicationGateway, cbCtx *appgw.ConfigBuilderContext, event events.Event) {
-	ingress, ok := event.Value.(*v1beta1.Ingress)
-	if !ok {
-		return
-	}
+func (c AppGwIngressController) updateIngressStatus(appGw *n.ApplicationGateway, cbCtx *appgw.ConfigBuilderContext, ingress *v1beta1.Ingress) {
+	ips := getIPs(appGw, c.azClient)
 
-	// check if this ingress is for AGIC or not, it might have been updated
-	if !k8scontext.IsIngressApplicationGateway(ingress) || !cbCtx.InIngressList(ingress) {
-		if err := c.k8sContext.UpdateIngressStatus(*ingress, ""); err != nil {
-			c.recorder.Event(ingress, v1.EventTypeWarning, events.ReasonUnableToUpdateIngressStatus, err.Error())
-		}
+	// determine what ipAddress to attach
+	usePrivateIP, _ := annotations.UsePrivateIP(ingress)
+	usePrivateIP = usePrivateIP || cbCtx.EnvVariables.UsePrivateIP == "true"
+
+	ipConf := appgw.LookupIPConfigurationByType(appGw.FrontendIPConfigurations, usePrivateIP)
+	if ipConf == nil {
 		return
 	}
 
-	// determine what ip to attach
-	usePrivateIP, _ := annotations.UsePrivateIP(ingress)
-	usePrivateIP = usePrivateIP || cbCtx.EnvVariables.UsePrivateIP == "true"
-	if ipConf := appgw.LookupIPConfigurationByType(appGw.FrontendIPConfigurations, usePrivateIP); ipConf != nil {
-		if ipAddress, ok := c.ipAddressMap[*ipConf.ID]; ok {
-			if err := c.k8sContext.UpdateIngressStatus(*ingress, ipAddress); err != nil {
-				c.recorder.Event(ingress, v1.EventTypeWarning, events.ReasonUnableToUpdateIngressStatus, err.Error())
+	if newIP, found := ips[ipResource(*ipConf.ID)]; !found {
+		for _, lbi := range ingress.Status.LoadBalancer.Ingress {
```

Comment:
we can range over `ingress.Status.LoadBalancer.Ingress` in the update funciton as well.

Reply:
Sounds good!